packages/cliqz-home-react/src/AddSpeedDial.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import ToggleDisplay from 'react-toggle-display';
+import cliqz from './Cliqz';
+
+class AddSpeedDial extends Component {
+  constructor(props) {
+    super(props);
+    this.freshtab = cliqz.freshtab;
+    this.state = {
+      show: true,
+      value: '',
+      errorDuplicate: false,
+      errorInvalid: false
+    }
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleClick(event) {
+    this.setState({
+      show: !this.state.show
+    });
+  }
+
+  handleChange(event) {
+    this.setState({value: event.target.value});
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const url = this.state.value && this.state.value.trim();
+    if(!url) {
+      return;
+    }
+
+    return this.freshtab.addSpeedDial(url).then((resp) => {
+      if(resp.error) {
+        if(resp.reason === 'duplicate'){
+          this.setState({
+            errorDuplicate: true
+          });
+        } else {
+          this.setState({
+            errorInvalid: true
+          });
+        }
+      } else {
+        this.setState({
+          value: ''
+        });
+        this.setState({
+          show: !this.state.show
+        });
+
+        this.props.addSpeedDial(resp)
+      }
+      
+    })
+  }
+
+  render() {
+    return (
+      <div>
+        <ToggleDisplay show={this.state.show}>
+          <a href="#" onClick={this.handleClick}>
+            <svg width="40px" height="40px" viewBox="0 0 40 40" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
```

Comment:
We could use svg tag if we want to change the fill or stroke of the image programmatically.